src/hotspot/share/classfile/classLoaderData.cpp

```diff
@@ -202,9 +202,9 @@ OopHandle ClassLoaderData::ChunkedHandleList::add(oop o) {
 
 int ClassLoaderData::ChunkedHandleList::count() const {
   int count = 0;
-  Chunk* chunk = _head;
+  Chunk* chunk = Atomic::load_acquire(&_head);
   while (chunk != nullptr) {
-    count += chunk->_size;
+    count += Atomic::load(&chunk->_size);
```

Comment:
I don't think this is necessary - there is no concurrent write of this chunk's size possible whilst we read it.

Reply:
Apologies, yes there is. We publish `_head` and then update its `_size` after the fact. That has a bad smell about it IMO.